### ISSUE TITLE
OP2 reading fails with RuntimeError: invalid analysis_code

### ISSUE THREAD

**Issue by kovalev-nnov** [SteveDoyle2/pyNastran#711]
Hi Steve! Thank you for this useful and free product!

I try to visualize solver results. There are OP2 files that I've got from DAT files by running Mystran solver (no errors).

PyNastran works well in reading of one (12x4 mesh) OP2 file:
> [2022-11-02 18:57:23] DEBUG : op2.py:542 combine=False 
> [2022-11-02 18:57:23] DEBUG : op2.py:543 -------- reading op2 with read_mode=1 (array sizing) -------- 
> [2022-11-02 18:57:23] INFO : op2_scalar.py:1672 op2_filename = 'D:/data/12-4_gzy/Test_12x4.OP2' 
> [2022-11-02 18:57:23] DEBUG : op2_reader.py:231 date = (3, 24, 21) 
> [2022-11-02 18:57:23] DEBUG : op2_reader.py:302 mode = 'msc' 
> [2022-11-02 18:57:23] DEBUG : op2_scalar.py:1850 table_name=b'OUGV1' 
> [2022-11-02 18:57:30] DEBUG : oug.py:168 parse_approach_code - approach_code=75 tCode=1 int3=0 isubcase=1
> so - analysis_code=7 device_code=5 table_code=1 sort_code=0
> (75 - 5) // 10 = 7 
> [2022-11-02 18:57:30] DEBUG : op2_scalar.py:1850 table_name=b'OPG1' 
> [2022-11-02 18:57:32] DEBUG : op2_scalar.py:1850 table_name=b'OQGV1' 
> [2022-11-02 18:57:33] DEBUG : op2_scalar.py:1850 table_name=b'OEF1X' 
> [2022-11-02 18:57:37] DEBUG : op2_scalar.py:1850 table_name=b'OES1X1' 
> 
but fails with the similar (24x8 mesh) OP2 file:
> [2022-11-02 19:06:43] DEBUG : op2.py:542 combine=False 
> [2022-11-02 19:06:43] DEBUG : op2.py:543 -------- reading op2 with read_mode=1 (array sizing) -------- 
> [2022-11-02 19:06:43] INFO : op2_scalar.py:1672 op2_filename = 'D:/data/24-8_gzy/Test_24x8.OP2' 
> [2022-11-02 19:06:43] DEBUG : op2_reader.py:231 date = (3, 24, 21) 
> [2022-11-02 19:06:43] DEBUG : op2_reader.py:302 mode = 'msc' 
> [2022-11-02 19:06:43] DEBUG : op2_scalar.py:1850 table_name=b'OUGV1' 
> [2022-11-02 19:06:48] DEBUG : oug.py:168 parse_approach_code - approach_code=235 tCode=1 int3=0 isubcase=1
> so - analysis_code=23 device_code=5 table_code=1 sort_code=0
> (235 - 5) // 10 = 23 
> [2022-11-02 19:06:48] ERROR : load_actions.py:300 Traceback (most recent call last):
> File "d:\Programs\pyNastran_dev\pyNastran\pyNastran\gui\qt_files\load_actions.py", line 297, in on_load_results
> load_function(out_filenamei)
> File "d:\Programs\pyNastran_dev\pyNastran\pyNastran\converters\nastran\gui\nastran_io.py", line 5030, in load_nastran_results
> model = self._load_nastran_results_str(results_filename, log)
> File "d:\Programs\pyNastran_dev\pyNastran\pyNastran\converters\nastran\gui\nastran_io.py", line 5122, in _load_nastran_results_str
> model.read_op2(op2_filename, combine=False)
> File "d:\Programs\pyNastran_dev\pyNastran\pyNastran\op2\op2.py", line 553, in read_op2
> table_names = OP2_Scalar.read_op2(self, op2_filename=op2_filename,
> File "d:\Programs\pyNastran_dev\pyNastran\pyNastran\op2\op2_interface\op2_scalar.py", line 1704, in read_op2
> table_names = self._read_tables(table_name)
> File "d:\Programs\pyNastran_dev\pyNastran\pyNastran\op2\op2_interface\op2_scalar.py", line 1870, in _read_tables
> op2_reader.read_results_table()
> File "d:\Programs\pyNastran_dev\pyNastran\pyNastran\op2\op2_interface\op2_reader.py", line 4909, in read_results_table
> self.read_results_table4()
> File "d:\Programs\pyNastran_dev\pyNastran\pyNastran\op2\op2_interface\op2_reader.py", line 4932, in read_results_table4
> self._read_subtables()
> File "d:\Programs\pyNastran_dev\pyNastran\pyNastran\op2\op2_interface\op2_reader.py", line 5132, in _read_subtables
> self._read_subtable_3_4(table3_parser, table4_parser, passer)
> File "d:\Programs\pyNastran_dev\pyNastran\pyNastran\op2\op2_interface\op2_reader.py", line 5224, in _read_subtable_3_4
> table3_parser(data, ndata)
> File "d:\Programs\pyNastran_dev\pyNastran\pyNastran\op2\tables\oug\oug.py", line 235, in _read_oug1_3
> raise RuntimeError(msg)
> RuntimeError: invalid analysis_code...analysis_code=23
> data={'_encoding': 'utf8', 'load_as_h5': False, 'h5_file': None, 'size': 4, 'is_msc': True, 'is_nasa95': False, 'table_name': b'OUGV1', 'approach_code': 235, 'isubcase': 1, 'table_code': 1, 'tCode': 1, 'sort_code': 0, 'sort_method': 1, 'device_code': 5, 'analysis_code': 23, 'sort_bits': [0, 0, 0], 'random_code': 0, 'format_code': 1, 'num_wide': 8, 'acoustic_flag': 0, 'thermal': 0}

Could you please give me some advice why this error could happened and how I can fix it .
Thanks in advance.

**Comment by SteveDoyle2** [comment]
It's a mystran bug.  I'm not sure on the limits, but if you make a smaller result, it'll work.  By small, I mean like 10 nodes/elements in a result max.

**Comment by kovalev-nnov** [comment]
Hi Steve! Is it a well known Mystran bug?
I have the same problem with the smaller result (6x2 mesh):

> [2022-11-08 09:34:00] DEBUG : op2.py:568 combine=False 
> [2022-11-08 09:34:00] DEBUG : op2.py:569 -------- reading op2 with read_mode=1 (array sizing) -------- 
> [2022-11-08 09:34:00] INFO : op2_scalar.py:1758 op2_filename = 'D:/data/6-2_akl/mystran_plate_6x2_Mesh_6_2.OP2' 
> [2022-11-08 09:34:00] DEBUG : op2_reader.py:264 date = (3, 24, 21) 
> [2022-11-08 09:34:00] DEBUG : op2_reader.py:342 mode = 'msc' 
> [2022-11-08 09:34:00] DEBUG : op2_scalar.py:1946 table_name=b'OUGV1' 
> [2022-11-08 09:34:00] DEBUG : op2_common.py:1763 parse_approach_code - approach_code=31 tCode=1 int3=0 isubcase=1
> so - analysis_code=3 device_code=1 table_code=1 sort_code=0
> 
> [2022-11-08 09:34:00] ERROR : load_actions.py:300 Traceback (most recent call last):
> File "D:\Programs\pyNastran_dev\pyNastran\pyNastran\gui\qt_files\load_actions.py", line 297, in on_load_results
> load_function(out_filenamei)
> File "D:\Programs\pyNastran_dev\pyNastran\pyNastran\converters\nastran\gui\nastran_io.py", line 5416, in load_nastran_results
> model = self._load_nastran_results_str(results_filename, log)
> File "D:\Programs\pyNastran_dev\pyNastran\pyNastran\converters\nastran\gui\nastran_io.py", line 5516, in _load_nastran_results_str
> model.read_op2(op2_filename, combine=False)
> File "D:\Programs\pyNastran_dev\pyNastran\pyNastran\op2\op2.py", line 579, in read_op2
> table_names = OP2_Scalar.read_op2(self, op2_filename=op2_filename,
> File "D:\Programs\pyNastran_dev\pyNastran\pyNastran\op2\op2_interface\op2_scalar.py", line 1793, in read_op2
> self._read_tables(table_name, table_names)
> File "D:\Programs\pyNastran_dev\pyNastran\pyNastran\op2\op2_interface\op2_scalar.py", line 1966, in _read_tables
> op2_reader.read_results_table()
> File "D:\Programs\pyNastran_dev\pyNastran\pyNastran\op2\op2_interface\op2_reader.py", line 6215, in read_results_table
> self.read_results_table4()
> File "D:\Programs\pyNastran_dev\pyNastran\pyNastran\op2\op2_interface\op2_reader.py", line 6240, in read_results_table4
> self._read_subtables()
> File "D:\Programs\pyNastran_dev\pyNastran\pyNastran\op2\op2_interface\op2_reader.py", line 6441, in _read_subtables
> self._read_subtable_3_4(table3_parser, table4_parser, passer)
> File "D:\Programs\pyNastran_dev\pyNastran\pyNastran\op2\op2_interface\op2_reader.py", line 6566, in _read_subtable_3_4
> table3_parser(data, ndata)
> File "D:\Programs\pyNastran_dev\pyNastran\pyNastran\op2\tables\oug\oug.py", line 170, in _read_oug1_3
> raise RuntimeError(msg)
> RuntimeError: invalid analysis_code...analysis_code=3
> data={'_encoding': 'utf8', 'load_as_h5': False, 'h5_file': None, 'size': 4, 'is_msc': True, 'is_nasa95': False, 'table_name': b'OUGV1', 'approach_code': 31, 'isubcase': 1, 'table_code': 1, 'tCode': 1, 'sort_code': 0, 'sort_method': 1, 'device_code': 1, 'analysis_code': 3, 'sort_bits': SortBits(is_complex=0, is_sort2=0, is_random=0), 'random_code': 0, 'format_code': 1, 'num_wide': 8, 'acoustic_flag': 0, 'thermal': 0}

TIA, Alex

**Comment by SteveDoyle2** [comment]
No, it's not well known.  I learned about it a few weeks ago and didn't have a bad model until a week ago.  It was a solid bar model that took a long time.  It failed on the displacements, so I deleted all the elements but 1 to run it faster.  Then I deleted nodes until it worked.  I think I had about 10 when it finally worked.  

I think it's buffer size dependent and the displacement table is ~1/5 of the shell stress table.  Try a 2x2.  Fails on large problems is an understatement.
